### PR TITLE
Add frontend panel file

### PR DESCRIPTION
**Pull request by UtsavSomaiya**
In this file,
Using java script, i am creating html elements and add to cart..

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -32,6 +32,104 @@
     <title>Retail Shop</title>
 
     <link rel="stylesheet" href="https://unpkg.com/tailwindcss@2.0.2/dist/tailwind.min.css">
+    <script type="text/javascript">
+        const cart = [];
+        function add_to_cart(id) {
```
Comment (gauravmak) [inline index.php:37]: addToCart

File: index.php [hunk 2]
```diff
@@ -32,6 +32,104 @@
...
+          const product = {
+            "id": id,
+            "name": document.getElementById('name-' + id).innerHTML,
+            "img": document.getElementById('image-' + id).src,
+            "price": document.getElementById('price-' + id).innerHTML,
+            "quantity" : 1
+          };
+          if (!add_to_array(cart, id)) {
```
Comment (gauravmak) [inline index.php:45]: existsInArray

File: index.php [hunk 3]
```diff
@@ -32,6 +32,104 @@
...
+            cart.push(product);
+          } else {
+            var index = cart.findIndex((obj => obj.name == document.getElementById('name-' + id).innerHTML));
```
Comment (gauravmak) [inline index.php:48]: compare id

File: index.php [hunk 4]
```diff
@@ -32,6 +32,104 @@
...
+            price = parseInt(price.slice(1));
+            price *= cart[index].quantity;
+            price = "$" + price;
+            cart[index].price = price;
+          }
+          displayCart(id);
+        }
+        function add_to_array(cart,id){
+           return cart.some(function(e1){
+             return e1.name === document.getElementById('name-' + id).innerHTML;
```
Comment (gauravmak) [inline index.php:60]: compare with id

Comment (gauravmak) [reply index.php:60]: what is e1?

File: index.php [hunk 5]
```diff
@@ -32,6 +32,104 @@
...
+
+
+            var div_main_sub_third = document.createElement('div');
+            div_main_sub_third.setAttribute('class', 'font-semibold text-lg w-16 text-center');
+            div_main_sub_third.innerHTML = product_price;
+            document.getElementById('div_main_' + id).appendChild(div_main_sub_third);
+
+            id++;
+          }
+          //input_value = 1;
```
Comment (gauravmak) [inline index.php:130]: ?

Comment (gauravmak) [inline index.php:130]: ?

File: index.php [hunk 6]
```diff
@@ -32,6 +32,104 @@
...
+            cart[index].price = price;
+          }
+          displayCart(id);
+        }
+        function add_to_array(cart,id){
+           return cart.some(function(e1){
+             return e1.name === document.getElementById('name-' + id).innerHTML;
+          });
+        }
+        function displayCart(id) {
```
Comment (gauravmak) [inline index.php:63]: No need to pass

**Review by gauravmak — CHANGES_REQUESTED** [review]

**Review by UtsavSomaiya — COMMENTED** [review]
All done

File: index.php [hunk 7]
```diff
@@ -32,6 +32,105 @@
...
+          });
+        }
+        function displayCart() {
+
+          document.getElementById('container').innerHTML = "";
+          id = 1;
+
+          for (let i = 0; i < cart.length; i++) {
+
+            var div_main = document.createElement('div');
```
Comment (gauravmak) [inline index.php:70]: Check variable naming camelcase

File: index.php [hunk 8]
```diff
@@ -32,6 +32,105 @@
...
+            var name_text = document.createTextNode(product_name);
+            span.appendChild(name_text);
+            document.getElementById('div_sub_first_' + id).appendChild(span);
+
+            var div_main_sub_second = document.createElement('div');
+            div_main_sub_second.setAttribute('class', 'w-32 flex justify-between');
+            div_main_sub_second.setAttribute('id', 'div_sub_second_' + id);
+            document.getElementById('div_main_' + id).appendChild(div_main_sub_second);
+
+            var button_1 = document.createElement('button');
```
Comment (gauravmak) [inline index.php:103]: ??

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: index.php (hunk 7 above)
Comment (gauravmak) [reply index.php:70]: ??

File: index.php (hunk 8 above)
Comment (gauravmak) [reply index.php:103]: What is button_1??

File: index.php [hunk 9]
```diff
@@ -32,6 +32,125 @@
...
+          inputValue = isNaN(inputValue) ? 0 : inputValue;
+          inputValue < 1 ? inputValue = 1 : '';
+          inputValue--;
+          var indexOfProduct = cart.findIndex((obj => obj.id == productId));
+          cart[indexOfProduct].quantity -= 1;
+          document.getElementById('input-id-'+id).value = inputValue;
+          price = parseInt((document.getElementById('price-'+id).innerHTML).slice(1)) * inputValue ;
+          document.getElementById('div-sub-third-'+id).innerHTML = '$'+price;
+          cart[indexOfProduct].price = '$'+price;
+        }
```
Comment (gauravmak) [inline index.php:152]: Can we combine these two?

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: index.php [hunk 10]
```diff
@@ -123,33 +125,30 @@ function displayCart() {
...
-          document.getElementById('input-id-'+id).value = inputValue;
-          price = parseInt((document.getElementById('price-'+id).innerHTML).slice(1)) * inputValue ;
-          document.getElementById('div-sub-third-'+id).innerHTML = '$'+price;
-          cart[indexOfProduct].price = '$'+price;
-        }
+          function increasingDecreasingQuantity(productId, id, checked) {
+            var inputValue = parseInt(document.getElementById('input-id-' + id).value, 10);
+            var indexOfProduct = cart.findIndex((obj => obj.id == productId));
+            if (checked == true) {
+              inputValue = isNaN(inputValue) ? 0 : inputValue;
```
Comment (gauravmak) [inline index.php:135]: This line is common right?

File: index.php [hunk 11]
```diff
@@ -123,33 +125,30 @@ function displayCart() {
...
-          inputValue < 1 ? inputValue = 1 : '';
-          inputValue--;
-          var indexOfProduct = cart.findIndex((obj => obj.id == productId));
-          cart[indexOfProduct].quantity -= 1;
-          document.getElementById('input-id-'+id).value = inputValue;
-          price = parseInt((document.getElementById('price-'+id).innerHTML).slice(1)) * inputValue ;
-          document.getElementById('div-sub-third-'+id).innerHTML = '$'+price;
-          cart[indexOfProduct].price = '$'+price;
-        }
+          function increasingDecreasingQuantity(productId, id, checked) {
```
Comment (gauravmak) [inline index.php:131]: changeQuantity

File: index.php [hunk 12]
```diff
@@ -101,7 +101,8 @@ function displayCart() {
 
             var decreaseButton = document.createElement('button');
             decreaseButton.setAttribute('class', 'px-3 py-1 rounded-md bg-gray-300');
-            decreaseButton.setAttribute('onclick','decreaseQuantity('+productId+','+id+')');
+            decreaseButton.setAttribute('id','decrease-button-'+id);
```
Comment (gauravmak) [inline index.php:104]: is this needed?

File: index.php [hunk 13]
```diff
@@ -101,7 +101,8 @@ function displayCart() {
...
+            decreaseButton.setAttribute('onclick','increasingDecreasingQuantity('+productId+','+id+','+false+')');
```
Comment (gauravmak) [inline index.php:105]: How about passing +1 or -1 instead of true or false?

**Review by gauravmak — CHANGES_REQUESTED** [review]

**Pull request merged**